src/librustc_trans/trans/_match.rs

```diff
@@ -1632,13 +1745,23 @@ pub fn store_arg<'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
                 // already put it in a temporary alloca and gave it up, unless
                 // we emit extra-debug-info, which requires local allocas :(.
                 let arg_val = arg.add_clean(bcx.fcx, arg_scope);
+
+                // We must zero indirect arguments, since they alias the original.
+                //
+                // FIXME (#5016): pnkfelix is not 100% sure how we are
+                // going handle this long-term. Maybe easiest to just
+                // not allow arguments that are Drop to be passed
+                // indirectly.
```

Comment:
If the caller is trying to drop the alloca used for an indirect argument, that's a bug.  They shouldn't be touched at all after a call because the callee is guaranteed to drop them.  (At least, that's how I understand the current model.)